File: BitBuffer-Java/src/com/fantaros/lib/bitbuffer/IBitBuffer.java
package com.fantaros.lib.bitbuffer;

import java.io.InputStream;

public interface IBitBuffer {

  byte peek(int offset);

  boolean peekOne(int offset);

  void zero();

  void fill();

  void write(String data, int startOffset, int bits);

  void write(byte[] data, int startOffset, int bits);

  void write(InputStream data, int startOffset, int bits);

  Bit range(int startOffset, int bits);

  byte[] getBytes(int startOffset, int bytes);
}

File: BitBuffer-Mono/BitBuffer-Mono/IBitBuffer.cs
using System;
using System.IO;
namespace com.fantaros.lib.bitbuffer
{
	public interface IBitBuffer
	{
		byte peek(int offset);
		bool peekOne(int offset);
		void enlarge(int bitsize);
		void empty();
		void fill();
		void write(string data,int startOffset,int bits);
		void write(byte[] data,int startOffset,int bits);
		void write(Stream data,int startOffset,int bits);
		byte[] range(int startOffset,int bits);
	}
}

